test/jobs/sync_track_job_test.rb

```diff
@@ -0,0 +1,10 @@
+require "test_helper"
+
+class SyncTrackJobTest < ActiveJob::TestCase
+  test "track is synced" do
+    track = create :track, slug: 'csharp'
+
+    Git::SyncTrack.expects(:call).with(track)
+    SyncTrackJob.perform_now(track)
+  end
```

Comment:
Is it synced now?

Reply:
I don't understand the question :)

Reply:
- The test is "test that track is sycned".
- The test expects a command to be called.

Questions:
- Are spies automatically verified at the end of the test?
- Is testing if the command is called the same as "the track is synced"? If yes, do we test the command? If no, is there a better name? _Gewetensvraag_: is it worth it to test an implementation detail like this? Or is this test too low-level to be meaningful?

Reply:
> Are spies automatically verified at the end of the test?

They are.

> If yes, do we test the command

We test that separately, yes.

> is it worth it to test an implementation detail like this? Or is this test too low-level to be meaningful?

🤷‍♂️ I've used the same attempt that was already taken in https://github.com/exercism/v3-website/blob/master/test/jobs/award_badge_job_test.rb

I do agree that the test name is slightly misleading.

Reply:
Yah. YMMV, I know that Jeremy also sometimes has tests like this because "things sometimes break", but I don't. Just a thought exercise, nothing blocking.

I'd personally test:

- the ref of the track is not up to date (invariant)
- perform the sync job
- the ref of the track is up to date (assert)

I really don't care if it's calling a certain command, but I do care that the job syncs the track.